sycl/include/sycl/ext/oneapi/properties/property_utils.hpp

```diff
@@ -106,93 +107,20 @@ template <typename RHS> struct SelectNonVoid<void, RHS> {
   using type = RHS;
 };
 
-// Merges two tuples by recursively extracting the type with the minimum
-// PropertyID in the two tuples and prepending it to the merging of the
-// remaining elements.
-template <typename T1, typename T2> struct Merge {};
-template <typename... LTs> struct Merge<std::tuple<LTs...>, std::tuple<>> {
-  using type = std::tuple<LTs...>;
+// Sort types accoring to their PropertyID.
+struct SortByPropertyId {
+  template <typename T1, typename T2>
+  using fn = sycl::detail::boost::mp11::mp_bool <
+             PropertyID<T1>::value<PropertyID<T2>::value>;
```

Comment:
Nit; Just to make it a little easier to read:

```suggestion
             (PropertyID<T1>::value < PropertyID<T2>::value)>;
```
Hopefully the formatter agrees with this.